src/common/utils.js

```diff
@@ -0,0 +1,33 @@
+export const deferred = () => {
+  let resolve;
+  let reject;
+  const promise = new Promise((presolve, preject) => {
+    resolve = presolve;
+    reject = preject;
+  });
+
+  return {
+    promise,
+    resolve,
+    reject
+  };
+};
+
+export const waitForGlobal = (name, intervalTime = 600, attempts = 5) => {
+  const defer = deferred();
+  let currentAttempt = 0;
+  let interval = setInterval(() => {
+    currentAttempt++;
+    if (global[name]) {
+      defer.resolve(global[name]);
+      clearInterval(interval);
```

Comment:
clearInterval() could be avoided by spawning a setTimeout() while the condition isn't met.

Reply:
Agree